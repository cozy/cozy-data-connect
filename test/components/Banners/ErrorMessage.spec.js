'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import { tMock } from '../../jestLib/I18n'
import { ErrorMessage } from 'components/Banners/ErrorMessage'

jest.mock('cozy-ui/transpiled/react/utils/color', () => ({
  getCssVariableValue: () => '#fff'
}))

const konnectorMock = {
  slug: 'mocK'
}

const errorMock = {
  type: 'LOGIN_FAILED',
  code: 'LOGIN_FAILED'
}

const twoFaExpiredErrorMock = {
  type: 'USER_ACTION_NEEDED',
  code: 'USER_ACTION_NEEDED.TWOFA_EXPIRED'
}

const twoFaWrongErrorMock = {
  type: 'USER_ACTION_NEEDED',
  code: 'USER_ACTION_NEEDED.WRONG_TWOFA_CODE'
}

const triggerMock = {}

const getComponent = error => {
  return shallow(
    <ErrorMessage
      konnector={konnectorMock}
      t={tMock}
      error={error}
      trigger={triggerMock}
    />
  )
    .shallow()
    .dive()
    .getElement()
}

describe('ErrorMessage component', () => {
  it(`Should render correctly`, () => {
    expect(getComponent(errorMock)).toMatchSnapshot()
  })

  it(`Should render correctly if two fa session expired`, () => {
    expect(getComponent(twoFaExpiredErrorMock)).toMatchSnapshot()
  })

  it(`Should render correctly if wrong twofa code error`, () => {
    expect(getComponent(twoFaWrongErrorMock)).toMatchSnapshot()
  })
})
