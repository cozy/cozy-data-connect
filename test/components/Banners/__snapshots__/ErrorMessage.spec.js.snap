// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component Should render correctly 1`] = `
<div
  className="Infos u-flex u-flex-column u-p-1 u-mih-2 u-flex-justify-center u-ta-left u-stack-m u-bg-chablis u-maw-none"
>
  <SubTitle
    className="u-pomegranate"
    ellipsis={false}
    tag="div"
  >
    Bad credentials
  </SubTitle>
  <div
    className="u-flex u-w-100"
  >
    <Text
      className="u-ta-left"
      ellipsis={false}
      tag="div"
    >
      Bad credentials. Check the konnector fields and run the connection again.
    </Text>
  </div>
</div>
`;

exports[`ErrorMessage component Should render correctly if two fa session expired 1`] = `
<div
  className="Infos u-flex u-flex-column u-p-1 u-mih-2 u-flex-justify-center u-ta-left u-stack-m u-bg-chablis u-maw-none"
>
  <SubTitle
    className="u-pomegranate"
    ellipsis={false}
    tag="div"
  >
    Relaunch the connexion to the service to fetch your data
  </SubTitle>
  <div
    className="u-flex u-w-100"
  >
    <Text
      className="u-ta-left"
      ellipsis={false}
      tag="div"
    >
      The last connexion to the service failed; please launch it again. You may have to provide a validation code.
    </Text>
  </div>
  <div
    className="u-flex-shrink-0"
  >
    <Wrapper
      trigger={Object {}}
    >
      [Function]
    </Wrapper>
  </div>
</div>
`;

exports[`ErrorMessage component Should render correctly if wrong twofa code error 1`] = `
<div
  className="Infos u-flex u-flex-column u-p-1 u-mih-2 u-flex-justify-center u-ta-left u-stack-m u-bg-chablis u-maw-none"
>
  <SubTitle
    className="u-pomegranate"
    ellipsis={false}
    tag="div"
  >
    Wrong two-factore code provided
  </SubTitle>
  <div
    className="u-flex u-w-100"
  >
    <Text
      className="u-ta-left"
      ellipsis={false}
      tag="div"
    >
      The two-factor code provided is wrong, please start again.
    </Text>
  </div>
  <div
    className="u-flex-shrink-0"
  >
    <Wrapper
      trigger={Object {}}
    >
      [Function]
    </Wrapper>
  </div>
</div>
`;
